data_processing/cleaning_demographic_data.Rmd

```diff
@@ -52,11 +50,412 @@ demo_dat %>%
+                                      str_detect(ethnicity, "mex|puerto|cuban") ~ "hispanic/latinx",
+                                      str_detect(ethnicity, "unknown|#|applicable|data not|client|missing|rent") ~ "unknown",
+                                      is.na(ethnicity) ~ "unknown",
+                                      
+                                      
+                                         TRUE ~ ethnicity))
+```
+
+check for ethnicity 
+```{r}
+demo_dat %>% 
+  count(ethnicity, race_2, ethnicity_recode) 
+```
+
+current_martial_status - replace unknowns and data not collected with "unknown". 
+
+### Question: How would we like to recode domestic partnership? (Lump with married?) What to do with single? (100/103 records with single are from LA) ALSO, what to do with "no"? (both from Merced, might get fixed when column shift issue is solved)
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(current_marital_recode = case_when(str_detect(current_marital_status, "unknown|#|applicable|data not") ~ "unknown",
+                                            str_detect(current_marital_status, "domestic|single") ~ "flag",
+                                      is.na(current_marital_status) ~ "unknown",
+                                         TRUE ~ current_marital_status))
+```
+
+check for marital status
+```{r}
+demo_dat %>% 
+  count(current_marital_status, current_marital_recode)
+```
+
+sexual_orientation - changed straight to "straight/heterosexual" to match CDSS variable. Changed all unknown or declined or missing responses to "unknown"
+
+### Question: do we want to lump "another sexual orientation" or "other" into something? Perhaps an "other sexual orientation" bucket?
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(sexual_orientation_recode = case_when(str_detect(sexual_orientation, "strai") ~ "straight/heterosexual",
+                                               str_detect(sexual_orientation, "unknown|#|client|decline|data not") ~ "unknown",
+                                               str_detect(sexual_orientation, "other") ~ "flag",
+                                      is.na(sexual_orientation) ~ "unknown",
+                                         TRUE ~ sexual_orientation))
+```
+
+check for sexual orientation
+```{r}
+demo_dat %>% 
+  count(sexual_orientation, sexual_orientation_recode)
+```
+
+preferred_language - leaving alone the responses that don't appear to be languages for now. Combining the mandarin/cantonese responses. Changing unknowns and missings to "unknown". 
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(preferred_language_recode = case_when(str_detect(preferred_language, "mandarin") ~ "mandarin/cantonese",
+                                               str_detect(preferred_language, "unknown|#|data not") ~ "unknown",
+                                               is.na(preferred_language) ~ "unknown",
+                                               TRUE ~ preferred_language))
+```
+
```

Comment:
Sounds good as above, we need to re-eval the distribution of the new data. Also, same comment as above, there are strings that start with "4" that should be recoded to `unknown`.